### ISSUE TITLE
[BUG] Twist.moe won’t work in IPads at all showing no data for searching or streaming an anime. 9anime as well just let you open the anime profile but not show epsoides

### ISSUE THREAD

**Issue by Moath022** [SuperMarcus/NineAnimator#223]
**STOP!!** Before submitting this bug report, make sure you have filled out the information below and renamed the title of this issue (ex. `[BUG] UrlError when trying to stream an episode on 9ani.app`). The information you provide helps us diagnose the problem and organize our issue tracker.

**Describe the bug**
Replace this line with a clear and concise description of what the bug is.

**To Reproduce**
Replace this with the steps to reproduce the behavior:

**Expected behavior**
Replace this line with a clear and concise description of what you expected to happen.

**Screenshots**

**Device:**
Replace the following with the device on which you're experiencing this issue.
 - Device: [e.g. iPhone Xs Max]
 - OS: [e.g. iOS 13.3]
 - Version: [e.g. v1.2-3 or commit hash]

**Crash Log**
```
If applicable, paste the crash log inside this block.
(Goto Settings.app -> Privacy -> Analytics & Improvements -> Analytics Data to find crash logs)
```

**Additional context**

**Comment by SuperMarcus** [comment]
This issue should have been addressed in the latest versions of NineAnimator. Please checkout the release page for more information.

**Comment by Moath022** [comment]
> Twist.moe should have been addressed in the latest versions of NineAnimator. We won't be fixing 9anime in the near future. Please checkout the release page for more information.

Thanks for the reply. Sadly I tried both build 4 and 5 on my iPad Pro 4th generation nothing show up. It works just fine in my iPhone X though. Basically for twist.moe it show that there is no result found on the search area. And if you click into epsoides for an anime in the main area it shows that search completed no result for X anime.

**Comment by SuperMarcus** [comment]
@Moath022 can you send a screenshot of the error message? Also, make sure you're searching in Romaji. The twist source afaik only supports Romaji when searching.

**Comment by uttiya10** [comment]
I can confirm that there is an issue with streaming video on twist.moe. However, this issue is present on both the website and NineAnimator, therefore there is nothing we can do to fix it.

I unfortunately cannot reproduce the search bug.

**Comment by Moath022** [comment]
,

> @Moath022 can you send a screenshot of the error message? Also, make sure you're searching in Romaji. The twist source afaik only supports Romaji when searching.

I did but the issue isn’t with the search even with the anime’s that are on the homepage.
![69E5CFE7-E80A-43E6-A1F1-CF3D1D60C401](https://user-images.githubusercontent.com/78132717/106303542-286cee80-6252-11eb-8b01-0dfbe816aa1a.png)
![9A479222-EDD4-4FD1-84D6-D84FFD9005A3](https://user-images.githubusercontent.com/78132717/106303548-299e1b80-6252-11eb-954f-5078ffdd1f90.png)
](url)

**Comment by Moath022** [comment]
I actually tried every version of the app hoping one could work but sadly it didn’t. So I remained in 1.2.7 build 1 to watch with aniwatch until the recent update. Btw there is an app doing the same functionality in AppStore already and charging 80$ for a year. Obviously no one would pay that much.

**Comment by Moath022** [comment]
> I can confirm that there is an issue with streaming video on twist.moe. However, this issue is present on both the website and NineAnimator, therefore there is nothing we can do to fix it.
> 
> I unfortunately cannot reproduce the search bug.

It does work on iPhones I tested build 1.2.6 and 1.2.7 except for build 2 and 3. So I don’t think anything is wrong with the website.

**Comment by uttiya10** [comment]
**Update on the streaming issue:**
Twist.moe has fixed their mobile website. However, they have changed their CDN URL's, therefore NineAnimator needs to be updated as well to use these new domains. I will re-open this issue until the app has been updated.

@Moath022 Regarding the search problem:
You are using an old version of NineAnimator which has this search problem. Please update

**Comment by Moath022** [comment]
> **Update on the streaming issue:**
> Twist.moe has fixed their mobile website. However, they have changed their CDN URL's, therefore NineAnimator needs to be updated as well to use these new domains. I will re-open this issue until the app has been updated.
> 
> @Moath022 **Regarding the search problem:**
> You are using an old version of NineAnimator which has this search problem. Please update

It is build 5 but I tried the previous version as well

**Comment by uttiya10** [comment]
@Moath022 Can i have a screenshot of the version screen? Go to NineAnimator Settings and click `About`

**Comment by Moath022** [comment]
> @Moath022 Can i have a screenshot of the version screen? Go to NineAnimator Settings and click `About`

![4C03344D-E1BF-46FD-9B57-1D58CB87F933](https://user-images.githubusercontent.com/78132717/106305726-f01adf80-6254-11eb-9dbb-91314518457a.png)
![3B4C2280-9636-46FB-845A-1D83317990CB](https://user-images.githubusercontent.com/78132717/106305729-f0b37600-6254-11eb-9b72-f502a3581e03.png)
![0F314869-2560-4224-8730-1DCC87F9DB82](https://user-images.githubusercontent.com/78132717/106305730-f14c0c80-6254-11eb-8cf4-cd5f08d34810.png)


My bad I have both of them installed so a duplicate this is the difference but still not working just different text

**Comment by uttiya10** [comment]
Ok, now THAT is strange. Can you try using the english name: "You and Me"?

**Comment by Moath022** [comment]
> Ok, now THAT is strange. Can you try using the english name: "You and Me"?

Sure sorry I was adding something you may look at for the enhancement I promise you will like it.

**Comment by Moath022** [comment]
![726EFB57-3449-4DC9-9AF5-B7DF28372217](https://user-images.githubusercontent.com/78132717/106307256-088bf980-6257-11eb-97cc-4c2eaa0b615b.png)

**Comment by Moath022** [comment]
Try it with any iPad ( mini, air) tested but maybe the iOS version but that is the last thing that could make it. I am on iOS 14.4 for the iPad Pro but the others are in 14.3

**Comment by uttiya10** [comment]
> Try it with any iPad ( mini, air) tested but maybe the iOS version but that is the last thing that could make it. I am on iOS 14.4 for the iPad Pro but the others are in 14.3

I doubt that the ios version makes any difference. I'll be honest, i'm a bit confused, because searching the anime works perfectly on my device. I'll try to see if other people are having the same issue as well.

**Comment by Moath022** [comment]
> > Try it with any iPad ( mini, air) tested but maybe the iOS version but that is the last thing that could make it. I am on iOS 14.4 for the iPad Pro but the others are in 14.3
> 
> I doubt that the ios version makes any difference. I'll be honest, i'm a bit confused, because searching the anime works perfectly on my device. I'll try to see if other people are having the same issue as well.


Wait did you try and IPad ? It works fine in my iPhone as well

**Comment by uttiya10** [comment]
> Wait did you try any IPad ? It works fine in my iPhone as well

No, I do not have an iPad. Thats why i'm asking other people if they have the same problem

**Comment by Moath022** [comment]
> > Wait did you try any IPad ? It works fine in my iPhone as well
> 
> No, I do not have an iPad. Thats why i'm asking other people if they have the same problem

Ic thanks

**Comment by uttiya10** [comment]
@Moath022 We made an update that should fix twist.moe. Can you please try this new version and check if your problem has been fixed?

Download ipa file here:
https://github.com/SuperMarcus/NineAnimator/suites/1984186641/artifacts/39513556

**Comment by Moath022** [comment]
Sadly it still not working. Just for confirmation this version has the same build (5). I tried searching for any anime and the same issue occurred.

> @Moath022 We made an update that should fix twist.moe. Can you please try this n ew version and check if your problem has been fixed?
> 
> Download ipa file here:
> https://github.com/SuperMarcus/NineAnimator/suites/1984186641/artifacts/39513556

**Comment by Moath022** [comment]
![30EBB6D5-83D7-4B8E-89C7-1AD7D9227429](https://user-images.githubusercontent.com/78132717/107136232-5230b000-68f9-11eb-99d1-a6201ff912a1.png)
![92DA40D2-E101-4940-8653-7EE852B12A53](https://user-images.githubusercontent.com/78132717/107136235-53fa7380-68f9-11eb-8b5e-e8cb69b47b2a.png)
![65E0CAE1-F4F1-42ED-A891-D5728FB3A740](https://user-images.githubusercontent.com/78132717/107136236-56f56400-68f9-11eb-8e4f-c7ee4761ccd6.png)
![A8B6D90A-B82B-432D-A120-90CAA1CB83FE](https://user-images.githubusercontent.com/78132717/107136237-56f56400-68f9-11eb-88cb-2f118cd1b842.jpeg)

**Comment by uttiya10** [comment]
Can you search “Kimi No Na Wa”? Notice the space! For some reason, twist.moe search is very strict on names!

**Comment by Moath022** [comment]
> Can you search “Kimi No Na Wa”? Notice the space! For some reason, twist.moe search is very strict on names!

I don’t think it is an issue with the the spelling nothing show up no matter what I write. 
![B94C55F0-EE57-4EE4-BE2D-202CF23F851E](https://user-images.githubusercontent.com/78132717/107155522-ca868800-6970-11eb-91ce-da29849fc976.png)
![9801DFEB-B10D-47BA-B531-F4202DC4E35D](https://user-images.githubusercontent.com/78132717/107155523-cbb7b500-6970-11eb-9798-8d8fb1134f71.png)
![0640E64B-E72D-402B-BEB8-A2C62CB58665](https://user-images.githubusercontent.com/78132717/107155536-dc682b00-6970-11eb-92e0-83d7e6149c41.png) 

The last picture is from my iPhoneX with the same build iOS 13.5

**Comment by Moath022** [comment]
So something should show up at least whether you spell correctly or not but in this case nothing tbh

**Comment by Moath022** [comment]
> Can you search “Kimi No Na Wa”? Notice the space! For some reason, twist.moe search is very strict on names!
Please see the reply I forgot to make it link with your message

**Comment by uttiya10** [comment]
@SuperMarcus Do you have an idea on what might be happening here, that causes the ipad to always return no search results?

**Comment by Moath022** [comment]
> @SuperMarcus Do you have an idea on what might be happening here, that causes the ipad to always return no search results?

I am still immature in developing or fixing an app tbh. I just know few things to modify some apps to my liking ^_ but I am wondering and it is a wild guess. Twist.moe has a use desktop mode for iPads so different than of mobile phones and this might not let it read the website or something ( it is a wild guess ) like I said I don’t know much yet

**Comment by Joseos123** [comment]
would like to say this happens for me but on animedao instead, and twist.moe works perfectly fine
